### DISCUSSION TITLE
rbenv install 2.7.1 install failed - MacOs Big Sur

### DISCUSSION THREAD

**Discussion by SVynnytska** [rbenv/ruby-build#1860] [Build failures]
rbenv install 2.7.1 and RUBY_CONFIGURE_OPTS=--with-openssl-dir=/opt/local rbenv install 2.7.1

BUILD FAILED (macOS 11.2.1 using ruby-build 20210119)
The Ruby openssl extension was not compiled.
The Ruby zlib extension was not compiled.
/Users/myuser/.rbenv/versions/2.7.1/lib/ruby/2.7.0/yaml.rb:3: warning: It seems your ruby installation is missing psych (for YAML output).
To eliminate this warning, please install libyaml and reinstall your ruby.
The Ruby yaml extension was not compiled.
ERROR: Ruby install aborted due to missing extensions
Configure options used:
--prefix=/Users/myuser/.rbenv/versions/2.7.1
--with-openssl-dir=/opt/local
--enable-shared
--with-readline-dir=/usr/local/opt/readline
CC=clang
CFLAGS=-I/usr/local/include/
LDFLAGS=-L/Users/myuser/.rbenv/versions/2.7.1/lib -L/usr/local/opt/openssl/lib
CPPFLAGS=-I/Users/myuser/.rbenv/versions/2.7.1/include -I/usr/local/opt/openssl/include

**Comment by cgibsonmm** [comment]
Same issue here. I have looked at a ton of workarounds but have not had any luck getting to a solution.

<img width="1121" alt="Screen Shot 2021-02-19 at 3 00 41 PM" src="https://user-images.githubusercontent.com/12194168/108561284-f805f680-72cb-11eb-8fd9-1edda00e4d5c.png">

**Comment by SVynnytska** [comment]
@cgibsonmm I finally managed to fix this by removing all gems running `sudo gem uninstall -aIx` and reinstalling `rbenv`

**Comment by cgibsonmm** [comment]
No luck with this solution. I am going to look into some of the other issues.

**Comment by Neutrollized** [comment]
Hi!  I had some issues and I recently got it working last night.  I put it in a gist [here](https://gist.github.com/Neutrollized/37841827940b28b27ec2e54abbbcc408) -- I hope it helps!

EDIT: a lot of the stuff outlined in there are things that I see you've already done (reinstalling CommandLineTools), so perhaps my CFLAGS and RUBY_CONFIGURE_OPTS will help

**Comment by jazz-mobility** [comment]
For me this command worked on macOS 11.6.
```
sudo CFLAGS="-Wno-error=implicit-function-declaration -Wcompound-token-split-by-macro" rbenv install
```
